qiita_db/util.py

```diff
@@ -863,7 +863,7 @@ def convert_to_id(value, table, text_col=None):
                                      % (value, table))
         # If there was a result it was a single row and and single value,
         # hence access to [0][0]
-        return _id[0][0]
+        return int(_id[0][0])
```

Comment:
Well, that is what is confusing. In which cases it was not returning an int? AFAIK it should always return an int. If not, there might be another bug in some other place, so "patching" it in this way is not the way to go...